File: backend/src/main/java/com/accesodatos/service/ProductService.java
package com.accesodatos.service;

import java.util.List;

import com.accesodatos.dto.product.ProductRequestDto;
import com.accesodatos.dto.product.ProductResponseDto;

public interface ProductService {

	List<ProductResponseDto> getAllProducts();
	ProductResponseDto getProductById(Long id);
	Boolean createProduct(ProductRequestDto dto);
	Boolean updateProduct(ProductRequestDto dto, Long id);
	Boolean deleteProduct(Long id);
	Boolean manageProductSate(Long productId, String state);
}

File: frontend/src/service/product.service.ts
import api from "../service/api";
import { ApiResponseDto } from "../types/api";
import { ProductRequest, ProductResponse } from "../types/Product";

const createProduct = async (productRequest: ProductRequest) => {
  const response = await api.post(`/products`, productRequest);
  return response.data;
};

const getAllProducts = async () => {
  const response = await api.get<ApiResponseDto<ProductResponse[]>>(
    `/products`
  );
  return response.data.data;
};

const getProductById = async (productId: number) => {
  const response = await api.get<ProductResponse>(`/products/${productId}`);
  return response.data;
};

const updateProduct = async (
  productId: number,
  productRequest: ProductRequest
) => {
  const response = await api.put<ProductResponse>(
    `/products/${productId}`,
    productRequest
  );
  return response.data;
};

const ProductService = {
  createProduct,
  getAllProducts,
  getProductById,
  updateProduct,
};

export default ProductService;
